src/transformers/modeling_utils.py

```diff
@@ -2065,6 +2065,30 @@ def framework(self) -> str:
         """
         return "pt"
 
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        # For BC we keep the original `config_class` definition in case
+        # there is a `config_class` attribute (e.g. remote code models),
+        # otherwise we derive it from the annotated `config` attribute.
+
+        # defined in this particular subclass
+        child_annotation = cls.__annotations__.get("config", None)
+        child_attribute = cls.__dict__.get("config_class", None)
+
+        # defined in the class (this subclass or any parent class)
+        full_annotation = get_type_hints(cls).get("config", None)
+        full_attribute = cls.config_class
+
+        # priority (subclass class_config -> subclass annotation -> global class_config -> global annotation)
```

Comment:
you probably want a small test for this!

Reply:
added in https://github.com/huggingface/transformers/pull/39132/commits/eb2335dd0a17c4e77b2547e3331716663ca37b91